src/Simulation/QCTraceSimulator/QCTraceSimulatorCore.cs

```diff
@@ -30,6 +30,13 @@ public class QCTraceSimulatorCoreConfiguration
         /// of the call graph.
         /// </summary>
         public uint CallStackDepthLimit = uint.MaxValue;
+
+        /// <summary>
+        /// Constrols if depth or width optimization is favored.
+        /// If set to true, resulting circuit is optimized for depth by discouraging qubit reuse.
+        /// If set to false, resulting circuit is optimized for width by encouraging qubit reuse.
+        /// </summary>
+        public bool OptimizeDepth = false;
     }
```

Comment:
I would call it OptimizeQubitReuseForDepth. Otherwise it is confusing. Reader immediately starts thinking about global algorithm optimization for depth.

Reply:
I'd rather keep it as OptimizeDepth. It is described here in the comments and in the documentation so confusion should be minimal.